java/ingest/ingest-runner/src/test/java/sleeper/ingest/IngestRecordsIT.java

```diff
@@ -61,551 +69,852 @@
 import static sleeper.ingest.testutils.IngestRecordsTestDataHelper.schemaWithRowKeys;
 
 class IngestRecordsIT extends IngestRecordsTestBase {
-    @Test
-    void shouldWriteRecordsSplitByPartitionLongKey() throws Exception {
-        // Given
-        StateStore stateStore = inMemoryStateStoreWithFixedPartitions(new PartitionsBuilder(schema)
-                .rootFirst("root")
-                .splitToNewChildren("root", "L", "R", 2L)
-                .buildList()
-        );
-
-        // When
-        long numWritten = ingestRecords(schema, stateStore, getRecords()).getRecordsWritten();
-
-        // Then:
-        //  - Check the correct number of records were written
-        assertThat(numWritten).isEqualTo(getRecords().size());
-        //  - Check StateStore has correct information
-        FileReferenceFactory fileReferenceFactory = FileReferenceFactory.from(stateStore);
-        List<FileReference> fileReferences = stateStore.getFileReferences().stream()
-                .sorted(Comparator.comparing(FileReference::getPartitionId))
-                .collect(Collectors.toList());
-        FileReference leftFile = fileReferences.get(0);
-        FileReference rightFile = fileReferences.get(1);
-        assertThat(fileReferences)
-                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("filename", "lastStateStoreUpdateTime")
-                .containsExactly(
-                        fileReferenceFactory.partitionFile("L", 1L),
-                        fileReferenceFactory.partitionFile("R", 1L));
-        //  - Read files and check they have the correct records
-        assertThat(readRecords(leftFile))
-                .containsExactly(getRecords().get(0));
-        assertThat(readRecords(rightFile))
-                .containsExactly(getRecords().get(1));
-        //  - Check quantiles sketches have been written and are correct (NB the sketches are stochastic so may not be identical)
-        AssertQuantiles.forSketch(getSketches(schema, leftFile.getFilename()).getQuantilesSketch("key"))
-                .min(1L).max(1L)
-                .quantile(0.0, 1L).quantile(0.1, 1L)
-                .quantile(0.2, 1L).quantile(0.3, 1L)
-                .quantile(0.4, 1L).quantile(0.5, 1L)
-                .quantile(0.6, 1L).quantile(0.7, 1L)
-                .quantile(0.8, 1L).quantile(0.9, 1L).verify();
-        AssertQuantiles.forSketch(getSketches(schema, rightFile.getFilename()).getQuantilesSketch("key"))
-                .min(3L).max(3L)
-                .quantile(0.0, 3L).quantile(0.1, 3L)
-                .quantile(0.2, 3L).quantile(0.3, 3L)
-                .quantile(0.4, 3L).quantile(0.5, 3L)
-                .quantile(0.6, 3L).quantile(0.7, 3L)
-                .quantile(0.8, 3L).quantile(0.9, 3L).verify();
-    }
+    @Nested
+    @DisplayName("Using one file per leaf file writing strategy")
+    class OneFilePerLeafStrategy {
```

Comment:
I'm not sure there's much value in testing both file writing strategies here, since the fact that we're using IngestRecords doesn't seem like it could affect how the files are written? It might be worth reverting these tests to save execution time and simplify the test suite?